### ISSUE TITLE
Remove two sources of truth for default trie cache size

### ISSUE THREAD

**Issue by matklad** [near/nearcore#7066]
However, the overall code does feel a bit "bolted on". I would like to fix the following things in the follow up: 

* make sure that default and overriden cache capacity is handled in the same way. It doesn't make sense that the former is hard-coded, but the latter is overridable
* Cleanup defaults in StoreConfig, it feels messy that the source of truth are awkward functions which exists solely for serde
* Replace `TrieFactory` with just `TrieConfig` -- we don't need doer object here, just a bag of values
* revisit the format of the config, 

  ```
  "trie_cache_capacities": [
    [
      {
        "version": 1,
        "shard_id": 2
      },
      2000000
    ]
  ]
  ```

  doesn't look like a good config format, something like `{ "1:2": "2000000" }` would'd be much user friendlier.

_Originally posted by @matklad in https://github.com/near/nearcore/issues/7022#issuecomment-1155004725_

**Comment by matklad** [comment]
cc @jakmeier I think this will be closed by one of your PRs

**Comment by jakmeier** [comment]
Solved by #7566 + #7578
